gobblin-modules/gobblin-helix/src/main/java/gobblin/metastore/ZkStateStore.java

```diff
@@ -81,6 +84,24 @@ public ZkStateStore(String connectString, String storeRootDir, boolean compresse
     propStore = new ZkHelixPropertyStore<byte[]>(connectString, serializer, storeRootDir);
   }
 
+  @Alias("zk")
+  public static class Factory implements StateStore.Factory {
```

Comment:
It is a bit of a subtlety but it can come to byte us later. We've been moving away from Properties and toward typesafe Config. Overall, the former supports the Properties file format with some caveats. If we don't use the alias but we decide to use the class FQN, it has to be something like: gobblin.metastore.ZkStateStore$Factory.

In properties format, one can write
state.store.type= gobblin.metastore.ZkStateStore$Factory

That does not work if we try to read this with typesafe Config since it will try to expand Factory as a variable reference. So we have to write 

state.store.type='gobblin.metastore.ZkStateStore$Factory'

Reading that as a Properties collection will include the quotes.

Instead, if we have a top level class gobblin.metastore.ZkStateStoreFactory, we can have 

state.store.type=gobblin.metastore.ZkStateStoreFactory

This is compatible with both.